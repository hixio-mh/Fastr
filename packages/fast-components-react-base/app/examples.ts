import AutoSuggestExamples from "../src/auto-suggest/examples.data";
export { AutoSuggestExamples };

import BadgeExamples from "../src/badge/examples.data";
export { BadgeExamples };

import BreadcrumbExamples from "../src/breadcrumb/examples.data";
export { BreadcrumbExamples };

import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CardExamples from "../src/card/examples.data";
export { CardExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import ContextMenu from "../src/context-menu/examples.data";
export { ContextMenu };

import ContextMenuItem from "../src/context-menu-item/examples.data";
export { ContextMenuItem };

import DialogExamples from "../src/dialog/examples.data";
export { DialogExamples };

import DividerExamples from "../src/divider/examples.data";
export { DividerExamples };

import HorizontalOverflowExamples from "../src/horizontal-overflow/examples.data";
export { HorizontalOverflowExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import ListboxExamples from "../src/listbox/examples.data";
export { ListboxExamples };

import ListboxItemExamples from "../src/listbox-item/examples.data";
export { ListboxItemExamples };

import NumberFieldExamples from "../src/number-field/examples.data";
export { NumberFieldExamples };

import ProgressExamples from "../src/progress/examples.data";
export { ProgressExamples };

import RadioExamples from "../src/radio/examples.data";
export { RadioExamples };

import SelectExamples from "../src/select/examples.data";
export { SelectExamples };

import SliderExamples from "../src/slider/examples.data";
export { SliderExamples };

import SliderTrackItemExamples from "../src/slider-track-item/examples.data";
export { SliderTrackItemExamples };

import TabsExamples from "../src/tabs/examples.data";
export { TabsExamples };

import TextAreaExamples from "../src/text-area/examples.data";
export { TextAreaExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
