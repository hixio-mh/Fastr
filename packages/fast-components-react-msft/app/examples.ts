import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CallToActionExamples from "../src/call-to-action/examples.data";
export { CallToActionExamples };

import CaptionExamples from "../src/caption/examples.data";
export { CaptionExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import Dialog from "../src/dialog/examples.data";
export { Dialog };

import DividerExample from "../src/divider/examples.data";
export { DividerExample };

import FlipperExample from "../src/flipper/examples.data";
export { FlipperExample };

import HeadingExample from "../src/heading/examples.data";
export { HeadingExample };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import MetatextExamples from "../src/metatext/examples.data";
export { MetatextExamples };

import ParagraphExamples from "../src/paragraph/examples.data";
export { ParagraphExamples };

import ProgressExamples from "../src/progress/examples.data";
export { ProgressExamples };

import SubheadingExamples from "../src/subheading/examples.data";
export { SubheadingExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
