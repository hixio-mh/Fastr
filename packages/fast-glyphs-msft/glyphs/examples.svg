<svg width="16" height="16" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="Canvas">
<g id="Group">
<g id="Vector">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.3 5.8L 2.3 3.9L 0.6 4.8L -3.8147e-07 3.8L 1.7 2.9L 0.1 2L 0.7 1L 2.2 1.9L 2.2 2.38419e-08L 3.3 2.38419e-08L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(10.3999 1.09998)"/>
</g>
<g id="Vector_2">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.2 5.8L 2.2 3.9L 0.6 4.8L 4.76837e-08 3.8L 1.6 2.9L 4.76837e-08 2L 0.6 1L 2.2 1.9L 2.2 -1.90735e-07L 3.3 -1.90735e-07L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(1.19995 10.2)"/>
</g>
<g id="Vector_3">
<path d="M 0 8L 0 0L 8 0L 8 8L 0 8ZM 1 1L 1 7L 7 7L 7 1L 1 1ZM 14.4 13.1L 16 14L 15.4 15L 13.8 14.1L 13.8 16L 12.7 16L 12.7 14.1L 11 15.1L 10.4 14.1L 12 13.2L 10.4 12.3L 11 11.3L 12.6 12.2L 12.6 10.3L 13.7 10.3L 13.7 12.2L 15.3 11.3L 15.9 12.3L 14.4 13.1Z"/>
</g>
</g>
</g>
</svg>