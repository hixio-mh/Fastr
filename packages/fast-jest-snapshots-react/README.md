# FAST Jest snapshots React

`@microsoft/fast-jest-snapshots-react` is a small utility library for creating snapshot Jest tests for React components.
Given a single component, the `generateSnapshots` function will iterate over a set of prop data and create a snapshot test for each `prop` instance.

## Usage

To use, import the `ISnapshotTestSuite` interface and the `generateSnapshots` function from `@microsoft/fast-jest-snapshots-react`:

```ts
import {generateSnapshots, ISnapshotTestSuite} from "@microsoft/fast-jest-snapshots-react";
```

Then, create you test data snapshot suite, where every index of `data` is a valid prop object for your `component`:

```ts
// in a Jest test file
import {MyReactComponent, IMyReactComponentProps} from "my-react-component.tsx";
import {ISnapshotTestSuite, generateSnapshots} from "@microsoft/fast-jest-snapshots-react";

const testSuite: ISnapshotTestSuite<IMyReactComponentProps> = {
    name: 'test component',
    component: MyCustomReactComponentConstructor,
    data: [
        {
            id: "test-id",
            text: "hello world",
            // other component props
        }
    ]
};

// Generate snapshots using the prop and component data you defined
generateSnapshots(testSuite);
```

### Snapshot descriptions
Descriptions can optionally be applied to any snapshot for providing context to the test and the expectation of the test.
To do this, simply put the `prop` object into an array where the first index is the snapshot description and the second index is the `prop` object:

```ts
// in a Jest test file
import {MyReactComponent, IMyReactComponentProps} from "my-react-component.tsx";
import {ISnapshotTestSuite, generateSnapshots} from "@microsoft/fast-jest-snapshots-react";

const testSuite: ISnapshotTestSuite<IMyReactComponentProps> = {
    // ...
    data: [
        [
            "should render with an id attribute",
            {
                id: "test-id",
                text: "hello world",
                // other component props
            }

        ]
    ]
};
```
