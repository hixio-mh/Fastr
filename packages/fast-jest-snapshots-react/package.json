{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "sideEffects": false,
  "version": "2.3.0",
  "author": {
    "name": "Microsoft",
    "url": "https://discord.gg/FcSNfg4"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "build": "tsc -p ./tsconfig.json",
    "prepare": "npm run build",
    "start": "webpack-dev-server",
    "test": "npm run tslint",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'",
    "watch": "npm run build -- -w --preserveWatchOutput"
  },
  "jest": {
    "testURL": "http://localhost",
    "transform": {
      "^.+\\.tsx?$": "ts-jest",
      "^.+\\.jsx?$": "babel-jest"
    },
    "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx"
    ]
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^2.2.0",
    "@types/jest": "^22.2.2",
    "@types/react": "^16.3.6",
    "jest": "^22.4.3",
    "react": "^16.5.2",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^3.0.1"
  },
  "dependencies": {
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
