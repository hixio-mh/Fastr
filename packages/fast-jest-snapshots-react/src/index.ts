import * as React from "react";
import * as renderer from "react-test-renderer";

/**
 * A set of snapshot test cases
 */
export type ISnapshotTestCases<T> = Array<ISnapshotTestCase<T>>;

/**
 * An prop object to supply for snapshots. Optionally, a
 * an array can be provided where the first index is a string representing
 * a snapshot description, with the second index being the prop data
 */
export type ISnapshotTestCase<T> = T | [string, T];

/**
 * An interface describing component example objects used for snapshots and component testing.
 */
export interface ISnapshotTestSuite<T> {
    /**
     * The name of the component
     */
    name: string;

    /**
     * The component constructor
     */
    component: React.ComponentClass<T>;

    /**
     * An array of prop instances for the component
     */
    data: ISnapshotTestCases<T>;

    /**
     * The JSON schema for the components data
     * @deprecated - this property is not required for snapshot testing and will be removed
     */
    schema?: any;

    /**
     * The detail view component data
     * @deprecated - this property is not required for snapshot testing and will be removed
     */
    detailData?: T;

    /**
     * Documentation for the component
     * @deprecated - this property is not required for snapshot testing and will be removed
     */
    documentation?: JSX.Element;
}

/**
 * Executes a single snapshot test given a component, component data, and a test title
 */
export function renderSnapshot<T>(data: T, component: React.ComponentClass<T>, title: string): void {
    test(title, (): void => {
        const renderedComponent: any = renderer.create(React.createElement(component, data));
        const componentJson: JSON = renderedComponent.toJSON();

        expect(componentJson).toMatchSnapshot();
    });
}

/**
 * Generate a set of snapshot tests given a snapshot suite
 */
export function generateSnapshots<T>(examples: ISnapshotTestSuite<T>): void {
    const data: ISnapshotTestCases<T> = examples.data;
    const component: React.ComponentClass<T> = examples.component;

    if (Array.isArray(data)) {
        data.forEach((example: ISnapshotTestCase<T>, index: number): void => {
            let title: string;
            let props: T;

            if (Array.isArray(example)) {
                title = `${examples.name} ${example[0]}`;
                props = example[1];
            } else {
                title = `${examples.name}: ${index}`;
                props = example;
            }

            renderSnapshot(props, component, title);
        });
    }
}
