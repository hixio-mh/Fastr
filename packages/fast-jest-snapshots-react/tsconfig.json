{
    "compilerOptions": {
        "moduleResolution": "node",
        "module": "commonjs",
        "target": "ES6",
        "baseUrl": ".",
        "declaration": true,
        "jsx": "react",
        "outDir": "./dist"
    },
    "include": [
        "src/**/*"
    ]
}
