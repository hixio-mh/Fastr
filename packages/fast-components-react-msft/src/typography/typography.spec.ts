import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";

describe("typography snapshots", (): void => {
    generateSnapshots(examples);
});
