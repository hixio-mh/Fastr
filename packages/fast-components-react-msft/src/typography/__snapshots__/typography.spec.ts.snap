// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typography snapshots Typography: 0 1`] = `
<p
  className="typography-0-1-1 typography__1-0-1-2"
>
  Typography example string
</p>
`;

exports[`typography snapshots Typography: 1 1`] = `
<h1
  className="typography-0-1-11 typography__1-0-1-12"
  data-sketch-symbol="Typographic level 1"
>
  Typography example string
</h1>
`;

exports[`typography snapshots Typography: 2 1`] = `
<h2
  className="typography-0-1-21 typography__2-0-1-23"
  data-sketch-symbol="Typographic level 2"
>
  Typography example string
</h2>
`;

exports[`typography snapshots Typography: 3 1`] = `
<h3
  className="typography-0-1-31 typography__3-0-1-34"
  data-sketch-symbol="Typographic level 3"
>
  Typography example string
</h3>
`;

exports[`typography snapshots Typography: 4 1`] = `
<h4
  className="typography-0-1-41 typography__4-0-1-45"
  data-sketch-symbol="Typographic level 4"
>
  Typography example string
</h4>
`;

exports[`typography snapshots Typography: 5 1`] = `
<h5
  className="typography-0-1-51 typography__5-0-1-56"
  data-sketch-symbol="Typographic level 5"
>
  Typography example string
</h5>
`;

exports[`typography snapshots Typography: 6 1`] = `
<h6
  className="typography-0-1-61 typography__6-0-1-67"
  data-sketch-symbol="Typographic level 6"
>
  Typography example string
</h6>
`;

exports[`typography snapshots Typography: 7 1`] = `
<span
  className="typography-0-1-71 typography__7-0-1-78"
  data-sketch-symbol="Typographic level 7"
>
  Typography example string
</span>
`;

exports[`typography snapshots Typography: 8 1`] = `
<caption
  className="typography-0-1-81 typography__8-0-1-89"
  data-sketch-symbol="Typographic level 8"
>
  Typography example string
</caption>
`;

exports[`typography snapshots Typography: 9 1`] = `
<p
  className="typography-0-1-91 typography__9-0-1-100"
  data-sketch-symbol="Typographic level 9"
>
  Typography example string
</p>
`;
